tests/activemq5-unit-tests/src/test/java/org/apache/activemq/ZeroPrefetchConsumerTest.java

```diff
@@ -157,9 +157,9 @@ public void testTwoConsumers() throws Exception {
       // now lets receive it
       MessageConsumer consumer1 = session.createConsumer(queue);
       MessageConsumer consumer2 = session.createConsumer(queue);
-      TextMessage answer = (TextMessage) consumer1.receiveNoWait();
+      TextMessage answer = (TextMessage) consumer1.receive(5000);
       assertEquals("Should have received a message!", answer.getText(), "Msg1");
-      answer = (TextMessage) consumer2.receiveNoWait();
+      answer = (TextMessage) consumer2.receive(5000);
```

Comment:
@clebertsuconic yes something is async. If you following session.commit() at server, it's async, which means after commit you can't expect the message is already at client side. You have to give it some time to allow messages to be delivered to the client. Using receiveNoWait() is not proper here.